src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -84,6 +88,44 @@ public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] Function
             }
         }
 
+        [HttpPost]
+        [Route("admin/warmup")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        [RequiresRunningHost]
+        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost, [FromServices] IScriptHostManager scriptHostManager, [FromServices] IHostIdProvider hostIdProvider)
+        {
+            // The function must be called warmup and should be of type warmup trigger
+            FunctionDescriptor function = scriptHost.GetFunctionOrNull(WarmUpConstants.FunctionName);
+            if (function != null && function.Metadata.FunctionContainsInputBinding(WarmUpConstants.WarmupTrigger))
+            {
+                try
+                {
+                    var hostStatus = new HostStatus
+                    {
+                        State = scriptHostManager.State.ToString(),
+                        Version = ScriptHost.Version,
```

Comment:
How is it expected that a warmup function would use this information? I don't see how it would. Since this becomes a contract, it'd be good to only send information we think they actually need.

Reply:
You are right in that version may not be as useful. But, I think that `State` of the function can definitely be of some use. Like say if app is offline, maybe a user would want to not initialize certain components.

I start with just adding "State" in the payload and as we find other useful properties, we can update this.